src/main/java/urchatbot/parser/Parser.java

```diff
@@ -0,0 +1,178 @@
+        switch (commandType) {
+        case "TODO":
+            return parseTodoCommand(command);
+        case "DEADLINE":
+            return parseDeadlineCommand(command);
+        case "EVENT":
+            return parseEventCommand(command);
+        case "LIST":
+            return parseListCommand(command);
+        case "MARK":
+            return parseMarkCommand(command);
+        case "UNMARK":
+            return parseUnmarkCommand(command);
+        case "CLEAR":
+            return parseClearCommand(command);
+        case "DELETE":
+            return parseDeleteCommand(command);
+        case "PRINT":
+            return parsePrintCommand(command);
+        case "FIND":
+            return parseFindCommand(command);
+        case "BYE":
+            return parseExitCommand(command);
+        default:
+            throw new URChatBotException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Categorises and returns subclass of Command type based on entered command.
+     *
+     * @param time Time in the format of "yyyy-MM-dd HH:mm" or "yyyy-MM-dd" or "HH:mm."
+     * @return a String of time in the format of "MMM d yyyy HH:mm" or "MMM d yyyy" or "HH:mm"
+     */
+    public static String changeTimeFormat(String time) {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+            LocalDateTime d = LocalDateTime.parse(time, formatter);
+            return d.format(DateTimeFormatter.ofPattern("dd MM yyyy HH:mm"));
+        } catch (DateTimeParseException e) {
+            try {
+                LocalDate d = LocalDate.parse(time);
+                return d.format(DateTimeFormatter.ofPattern("dd MM yyyy"));
+            } catch (DateTimeParseException err) {
+                try {
+                    LocalTime t = LocalTime.parse(time);
+                    return t.format(DateTimeFormatter.ofPattern("HH:mm"));
+                } catch (DateTimeParseException tErr) {
+                    return null;
+                }
+            }
+        }
+    }
+
+
+    private static String getCommandType(String command) {
+        return command.split("\\s")[0].toUpperCase();
+    }
+
+    private static Command parseTodoCommand(String command) throws URChatBotException {
+        if (command.length() <= 5) {
+            throw new URChatBotException("OOPS!!! The description of a todo cannot be empty.");
+        }
+        String task = command.substring(command.indexOf("todo") + 5);
+        return new TodoCommand(task);
+    }
+    private static Command parseDeadlineCommand(String command) throws URChatBotException {
+        if (command.length() <= 5) {
+            throw new URChatBotException("OOPS!!! The description of a deadline cannot be empty.");
+        }
+        if (!command.contains("/by")
+                || command.substring(command.indexOf("deadline") + 9, command.indexOf("/by")).trim().length() < 1
```

Comment:
can consider shortening the booleans here. Perhaps make it into 1 boolean expression at the start as seen in lecture?